.gitlab/build.yml

```diff
@@ -141,47 +134,43 @@ build-adp-image-internal:
     - if: !reference [.on_mq_branch, rules, if]
       when: never
     - when: always
-  needs:
-    - build-adp-image-internal-amd64
-    - build-adp-image-internal-arm64
+      needs:
+        - build-adp-image-internal-amd64
+        - build-adp-image-internal-arm64
   variables:
     IMAGE_TAG: ${ADP_FULL_IMAGE_TAG}-internal
 
 build-adp-image-fips-internal-amd64:
-  extends: [.build-common-variables, .build-adp-definition]
+  extends: [.build-common-variables, .build-amd64-variables, .build-adp-definition]
   tags: ["docker-in-docker:amd64"]
   rules:
     - if: !reference [.on_mq_branch, rules, if]
       when: never
     - when: always
   variables:
-    ARCH_TAG: amd64
     APP_IMAGE: ${ADP_INTERNAL_BASE_IMAGE}
     BUILD_FEATURES: "fips"
-    FIPS_ENABLED: "true"
 
 build-adp-image-fips-internal-arm64:
-  extends: [.build-common-variables, .build-adp-definition]
+  extends: [.build-common-variables, .build-arm64-variables, .build-adp-definition]
   tags: ["docker-in-docker:arm64"]
   rules:
     - if: !reference [.on_mq_branch, rules, if]
       when: never
     - when: always
   variables:
-    ARCH_TAG: arm64
     APP_IMAGE: ${ADP_INTERNAL_BASE_IMAGE}
     BUILD_FEATURES: "fips"
-    FIPS_ENABLED: "true"
 
 build-adp-image-fips-internal:
   extends: [.build-common-variables, .build-adp-multi-arch-definition]
   rules:
     - if: !reference [.on_mq_branch, rules, if]
       when: never
     - when: always
-  needs:
-    - build-adp-image-fips-internal-amd64
-    - build-adp-image-fips-internal-arm64
+      needs:
+        - build-adp-image-fips-internal-amd64
+        - build-adp-image-fips-internal-arm64
```

Comment:
The various spots here where we're moving the `needs` dictionary under the `rules` stuff above is that without doing this, it essentially lets the job run as soon as the `needs` jobs finish... even if they _fail_.

I don't know why the normal behavior of them having to  pass is ignored when `rules` comes into play, but this shift of where we apply the `needs` solves the problem. 🤷🏻